test/e2e/framework/service/jig.go

```diff
@@ -858,15 +858,20 @@ func testReachabilityOverNodePorts(nodes *v1.NodeList, sp v1.ServicePort, pod *v
 			return err
 		}
 	}
-	for _, externalAddr := range externalAddrs {
-		isNodeExternalIPV4 := net.ParseIP(externalAddr).To4() != nil
-		if isClusterIPV4 != isNodeExternalIPV4 {
-			framework.Logf("skipping testEndpointReachability() for external adddress %s as it does not match clusterIP (%s) family", externalAddr, clusterIP)
-			continue
-		}
-		err := testEndpointReachability(externalAddr, sp.NodePort, sp.Protocol, pod)
-		if err != nil {
-			return err
+
+	// By default, skip node port checking with node external ips
+	if os.Getenv("CHECK_NODEPORT_WITH_EXTERNALADDR") == "true" {
```

Comment:
let me ask about this, I´m not sure how @BenTheElder @spiffxp prefer to parameterise these things?

Reply:
I'm not sure whether it is rude to remove this directly.

BTW, we should decide the default behavior: skip or check?

Reply:
ah yeah, but there is a miriad of flags, skips, checks, ... just what I prefer to wait for them so we can align all the test to behave the same

Reply:
I think this is usually done with a flag? though this seems awfully specific.

Also conformance tests should not have variable behavior, so IMO the correct answer is actually to split this out into another test and give it tag to filter on if we're keeping this test.

Changes to conformance tests behavior should also be approved by the conformance subproject.

Reply:
so like `[Feature:ExternalNodeAccess]` on the new tests or something.

But also why are we changing this? commenting back on the issue.
Just because some clusters can fail it doesn't necessarily mean we change conformance 😕

Reply:
for the PR, I will update as you suggest later after the holiday(maybe a week later😄)

for the issue, we can disscuss it in #90764.